File: MovileNext_Aula1/app/src/main/java/br/panaggio/movilenext_aula1/MainActivity.java
package br.panaggio.movilenext_aula1;

import android.arch.lifecycle.ViewModelProviders;
import android.os.SystemClock;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Chronometer;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ChronometerViewModel chronometerViewModel =
                ViewModelProviders.of(this).get(ChronometerViewModel.class);

        if (chronometerViewModel.getStartTime() == null) {
            long startTime = SystemClock.elapsedRealtime();
            chronometerViewModel.setStartTime(startTime);
        }

        Chronometer chronometer = findViewById(R.id.chronometer);
        chronometer.setBase(chronometerViewModel.getStartTime());
        chronometer.start();
    }
}

File: MovileNext_Aula3_DI/app/src/main/java/br/panaggio/movilenext_aula3_di/MainActivity.kt
package br.panaggio.movilenext_aula3_di

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import br.panaggio.movilenext_aula3_di.injection.DaggerTwitterComponent
import br.panaggio.movilenext_aula3_di.injection.NetworkModule
import br.panaggio.movilenext_aula3_di.injection.TwitterModule

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val component = DaggerTwitterComponent.builder()
                .twitterModule(TwitterModule("Bruna Panaggio"))
                .build()

        val twetter = component.tweeter()
        val module = component.timeline()
    }
}
